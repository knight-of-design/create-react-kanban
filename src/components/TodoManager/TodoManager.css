.Todo-manager {
  text-align: center;
  padding-top:15px;
}

.progress {
  position: absolute;
  right: 0;
}

.new-todo{
  font-size: 1.4em;
  padding: 7px;
  border-radius: 10px;
  width: 400px;
}

.todo-list{
  text-align:left;
  width: 400px;
  margin: 20px auto;
  list-style: none;

}

.todo-list li {
  padding: 6px;
}

.todo-list [type=checkbox] {
  margin-right: 8px;
}



