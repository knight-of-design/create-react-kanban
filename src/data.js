import TodoList from './models/TodoList'
import TaskList from './models/TaskList'

const taskList = new TaskList()
const taskLists = {}

for (let i = 0; i < 4; i++){
    const taskList = new TaskList()
    taskList.addTask('Do Chores')
    taskList.addTask('Travel to Hawaii')
    taskLists['taskList' + (i+1)] =  taskList


}


export default taskLists