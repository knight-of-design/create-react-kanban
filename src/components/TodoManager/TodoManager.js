import React, { Component } from 'react';
import './TodoManager.css';
import {observer} from 'mobx-react'

@observer
class TodoManager extends Component {

    render() {
        const { todoList } = this.props

        return (
            <div className="Todo-manager">
        <span className="progress">
         Progress: { todoList.progress }
        </span>
                <input className="new-todo" placeholder="Add Something To Do"  onKeyPress={this.onMakeTodo}/>

                <ul className="todo-list">
                    { todoList.todos.map((todo, index) => <Todo todo={todo} key={index} />)  }
                </ul>
            </div>
        );
    }

    onMakeTodo = (e) => {
        if (e.key==='Enter'){
            const todo = e.target.value
            this.props.todoList.addTodo(todo)
            e.target.value=''
        }
    };
}


@observer
class Todo extends React.Component {
    render() {
        const { todo } = this.props
        return (
            <li onDoubleClick={this.onRename}>
                <input
                    className="toggle"
                    type="checkbox"
                    checked={ todo.completed }
                    onChange={ this.onToggleCompleted }
                />
                <label>
                    { todo.task }
                </label>
            </li>
        );
    }
    onToggleCompleted = () => {
        const todo = this.props.todo
        todo.completed = !todo.completed
    };
    onRename = () => {
        const todo = this.props.todo
        todo.task = prompt('Task name', todo.task) || ""
    };
}

export default TodoManager;
