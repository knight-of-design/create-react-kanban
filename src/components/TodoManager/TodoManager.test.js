import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import TodoManager from './TodoManager';
import data from '../../data'

test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<TodoManager todoList={data.todoList} />, div);
    ReactDOM.unmountComponentAtNode(div);
});

test('renders a todo list', () => {

    const component = renderer.create(<TodoManager todoList={data.todoList} />)
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});