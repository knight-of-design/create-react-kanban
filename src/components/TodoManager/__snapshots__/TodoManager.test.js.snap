// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a todo list 1`] = `
<div
  className="Todo-manager"
>
  <span
    className="progress"
  >
    Progress: 
    0%
  </span>
  <input
    className="new-todo"
    onKeyPress={[Function]}
    placeholder="Add Something To Do"
  />
  <ul
    className="todo-list"
  >
    <li
      onDoubleClick={[Function]}
    >
      <input
        checked={false}
        className="toggle"
        onChange={[Function]}
        type="checkbox"
      />
      <label>
        Do Chores
      </label>
    </li>
    <li
      onDoubleClick={[Function]}
    >
      <input
        checked={false}
        className="toggle"
        onChange={[Function]}
        type="checkbox"
      />
      <label>
        Travel to Hawaii
      </label>
    </li>
  </ul>
</div>
`;
